OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs

```diff
@@ -431,9 +435,24 @@ void RemoveShellmapUI()
 			rootMenu.Parent.RemoveChild(rootMenu);
 		}
 
+		/// <summary>
+		/// If LastModifiedMap was picked already, returns a null
+		/// </summary>
+		public static string PickLastModifiedMap(ModData modData)
```

Comment:
`static` for `lastLoadedLastModifiedMap` and this method being `static` makes it feel like this method is living in the wrong place.

It looks like we're digging through state contained in `modData.MapCache` and that suggests to me that this method should live there instead?

Reply:
good idea

Reply:
This got me thinking, should we make more of MapCache static? As we have just one MapCache instance at a time we could make most public functions and variables static

Reply:
Prefer to keep it non-static.

Reply:
ok